### PR TITLE
Deprecate Outdated Launch Method

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - Deprecate `launchDropInResult(Activity, Int)` in favor of `launchDropInResult`
 - Addresses comments in issue #313

 ### Checklist

 - [x] Added a changelog entry

- @sshropshire

### CODE REVIEW

File: Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java [hunk 1]
```diff
@@ -486,6 +488,33 @@ public void launchDropInForResult(FragmentActivity activity, int requestCode) {
...
+     *
+     * @see #DropInClient(Fragment, DropInRequest, String)
+     * @see #DropInClient(Fragment, DropInRequest, ClientTokenProvider)
+     * @see #DropInClient(FragmentActivity, DropInRequest, String)
+     * @see #DropInClient(FragmentActivity, DropInRequest, ClientTokenProvider)
+     */
+    public void launchDropIn() {
+        getAuthorization((authorization, authorizationError) -> {
+            if (authorization != null) {
+                if (observer != null) {
```
Comment (jaxdesmarais) [inline Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java:505]: Is there a way to combine these nested if's into an `&&` to just single line and simplify it a bit?

File: Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java [hunk 2]
```diff
@@ -486,6 +488,33 @@ public void launchDropInForResult(FragmentActivity activity, int requestCode) {
...
+                    DropInIntentData intentData =
+                            new DropInIntentData(dropInRequest, authorization, braintreeClient.getSessionId());
+                    observer.launch(intentData);
+                }
+            } else if (authorizationError != null) {
+                if (listener != null) {
```
Comment (jaxdesmarais) [inline Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java:511]: Same question here. If it's not possible totally fine, just curious.

**Review by jaxdesmarais — APPROVED** [review]
⛵

File: Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java (hunk 1 above)
Comment (sshropshire) [reply Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java:505]: Yeah definitely that's a good call. I'll simplify the nesting structure here.

File: Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java (hunk 2 above)
Comment (sshropshire) [reply Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java:511]: +1

**Review by mattwylder — APPROVED** [review]
🤠

**Pull request merged**